frontend/src/app/containers/ExperimentPage/MobileDialog.js

```diff
@@ -0,0 +1,299 @@
+import { Localized } from "fluent-react/compat";
+import LocalizedHtml from "../../components/LocalizedHtml";
+import Loading from "../../components/Loading";
+
+import { subscribeToBasket, subscribeToBasketSMS, acceptedSMSCountries } from "../../lib/utils";
+
+const COUNTRY_CODE_ENDPOINT = "https://www.mozilla.org/country-code.json";
+
+/* TODO:
+- fix "mobile experiment" logic to include "ios" and "android" when only 1 is present.
+- fix input localization
+*/
+
+type MobileDialogProps = {
+  getWindowLocation: Function,
+  onCancel: Function,
+  sendToGA: Function,
+  experiment: Object
+}
+
+type MobileDialogState = {
+  loading: boolean,
+  isSuccess: boolean,
+  isError: boolean,
+  allowSMS: boolean,
+  submitAttempted: boolean,
+  recipient: string,
+  country: string
+}
+
+const DEFAULT_STATE = {
+  loading: true,
+  isSuccess: false,
+  isError: false,
+  allowSMS: false,
+  submitAttempted: false,
+  recipient: "",
+  country: "US"
+};
+
+export default class MobileDialog extends React.Component {
+  props: MobileDialogProps
+  state: MobileDialogState
+
+  modalContainer: Object
+
+  constructor(props: MobileDialogProps) {
+    super(props);
+    this.state = DEFAULT_STATE;
+  }
+
+  componentDidMount() {
+    this.focusModalContainer();
+    this.fetchCountryCode();
+  }
+
+  fetchCountryCode() {
+    // we use XMLHttpRequest here over fetch, because of issues with
+    // the country code endpoint not being parse correctly with
+    // fetch() for some reason I haven't figured out yet.
+    const req = new XMLHttpRequest();
+    req.onerror = (e) => {
+      this.setState({ loading: false });
+    };
+
+    req.onload = (e) => {
+      // $FlowFixMe - EventTarget type is missing response property
+      const country = e.target.response.country_code;
+      if (acceptedSMSCountries.includes(country)) {
+        this.setState({
+          loading: false,
+          allowSMS: true,
+          // $FlowFixMe - EventTarget type is missing response property
+          country: e.target.response.country_code
+        });
+      } else this.setState({ loading: false });
+    };
+    req.open("GET", COUNTRY_CODE_ENDPOINT);
+    req.responseType = "json";
+    req.send();
+  }
+
+  render() {
+    const { experiment } = this.props;
+    const { title, android_url, ios_url } = experiment;
+    const { isSuccess, allowSMS, loading } = this.state;
+
+    const headerMessage = ios_url ? (<LocalizedHtml id="mobileDialogMessageIOS" $title={title}>
+      <p>Download <b>{title}</b> from the iOS App Store.</p></LocalizedHtml>) : (<LocalizedHtml id="mobileDialogMessageAndroid" $title={title}><p>Download <b>{title}</b> from the Google Play Store.</p></LocalizedHtml>);
+
+    const headerImg = ios_url ? (<a href={ios_url} target="_blank"><img className="mobile-header-img" src="/static/images/ios.svg"/></a>) : (<a href={android_url} target="_blank"><img className="mobile-header-img" src="/static/images/google-play.png"/></a>);

```

Comment:
This is looking real slick. 

Question: is it just me or is the ios svg "fill" values not appearing? In other words, the button is a full black square and when I inspect the svg, Fx Developer Edition shows `style=""` instead of `style="fill: #fff"`.

![screen shot 2018-06-20 at 12 41 12 pm](https://user-images.githubusercontent.com/49511/41678128-7b3e2ac2-7487-11e8-917a-4ce63a760fb2.png)


It doesn't seem unique to this branch, as I see it on `master` too, but wanted to see if you thought it's worth looking into here...

Reply:
@devinreams did you set ios_url in the experiment's yaml file?

Reply:
That's the only thing I could think of that might cause that, as I see it locally

Reply:
@meandavejustice let's file a separate issue for this. i think i know what's going on.

Reply:
>Actually I'm seeing this too now

@meandavejustice ok, thanks for confirming I'm not _too_ crazy 😅 

>let's file a separate issue for this. i think i know what's going on.

@johngruen nice, please tell me it has to do with CSS 🎨

Reply:
more or less, yeah. inlining svg styles usually fixes this issue which we can do with svgo. I see it a lot with SVGs generated from adobe stuff.